src/services/signer_service.ts

```diff
@@ -173,12 +214,86 @@ export class SignerService {
                     shouldValidate: false,
                 },
             );
-
+        await this._updateTransactionEntityToSubmittedAsync(txHash);
         return {
             ethereumTransactionHash,
             signedEthereumTransaction,
         };
     }
+    private async _checkForStuckTransactionsAsync(): Promise<void> {
+        const stuckTransactions = await this._transactionEntityRepository.find({
+            where: { status: TransactionStates.Stuck },
+        });
+        if (stuckTransactions.length === 0) {
+            return;
+        }
+        const gasStationPrice = await this._getGasPriceFromGasStationOrThrowAsync();
+        const targetGasPrice = gasStationPrice.multipliedBy(UNSTICKING_TRANSACTION_GAS_MULTIPLIER);
+        for (const tx of stuckTransactions) {
+            try {
+                await this._unstickTransactionAsync(tx, targetGasPrice.multipliedBy(targetGasPrice));
+            } catch (err) {
+                logger.error(`failed to unstick transaction ${tx.hash}`, { err });
+            }
+        }
+    }
+    private async _unstickTransactionAsync(tx: TransactionEntity, gasPrice: BigNumber): Promise<string> {
+        const ethereumTxnParams: PartialTxParams = {
+            from: META_TXN_RELAY_ADDRESS,
+            to: META_TXN_RELAY_ADDRESS,
+            value: web3WrapperUtils.encodeAmountAsHexString(0),
+            nonce: web3WrapperUtils.encodeAmountAsHexString(tx.nonce),
+            chainId: CHAIN_ID,
+            gasPrice: web3WrapperUtils.encodeAmountAsHexString(gasPrice),
+            gas: web3WrapperUtils.encodeAmountAsHexString(ETH_TRANSFER_GAS_LIMIT),
+        };
+        const { signedEthereumTransaction, txHash } = this._getTransactionHashAndRawTxString(ethereumTxnParams);
+        const transactionEntity = TransactionEntity.make({
+            hash: txHash,
+            status: TransactionStates.Unsubmitted,
+            nonce: tx.nonce,
+            gasPrice,
+            metaTxnRelayerAddress: META_TXN_RELAY_ADDRESS,
+            expectedMinedInSec: EXPECTED_MINED_SEC,
+        });
+        await this._transactionEntityRepository.save(transactionEntity);
+        await this._web3Wrapper.sendRawPayloadAsync({
+            method: 'eth_sendRawTransaction',
+            params: [signedEthereumTransaction],
+        });
+        await this._updateTransactionEntityToSubmittedAsync(txHash);
+        return txHash;
```

Comment:
Why doesn't this mark the stuck tx as `Aborted`?  I see `transaction_watcher_service` attempts this by setting all stuck tx with the same nonce to aborted, but that feels like a catch all. Where setting it here would potentially be more targetted?

Reply:
Currently, we consider transactions `Aborted` ONLY when a transaction with the same nonce is `Confirmed`, whereas here we only know that the transaction was only submitted.